### ISSUE TITLE
Make core bundled extensions available (PDO, curl, intl, etc.)

### ISSUE THREAD

**Issue by asgrim** [php/pie#133]
Ref: https://bsky.app/profile/nizzness.bsky.social/post/3lbntufzx6c2i

> Anyone know if the pie installer has plans to add things like intl, pdo, curl etc...?

Extensions in the core such as PDO, curl, intl currently aren't available in PIE. Discuss how to make them available, since a root composer.json won't really work, as it's essentially one repo with a whole bunch of extensions.

**Comment by asgrim** [comment]
Couple of suggestions came out of PHPF discussions, I need to investigate viability further;

 - git submodules (very invasive on php-src repo, probably not ideal option...) 
 - sub-tree splits (like Symfony); possible in theory, would need a repo per ext
 - may be possible internally in PIE to define a custom repository which pulls from php-src

**Comment by derickr** [comment]
I don't think having core extensions in PIE makes sense. Most people will install these from distributions, and if they do choose to compile their own PHP, they have all the source to compile these extensions too.

**Comment by asgrim** [comment]
I must admit, I'm now struggling to understand when you'd `pie install intl` for example. I'd previously thought this was being requested for things like Dockerfile images, where the PHP install is fairly barebones, but @derickr pointed out to me that usually these images use their own distro packages, (e.g. deb.sury for Ubuntu, [`docker-php-ext-install`](https://github.com/docker-library/php/blob/master/docker-php-ext-install), Remi repo, etc.).

If anyone has a specific use-case for this please comment about it, we could still look into it, but I'm inclined to de-prioritise this at the moment if no further feedback is given. :thinking:

**Comment by macintoshplus** [comment]
On Dockerfile, you can use `docker-php-ext-install` to install a bundled PHP extension not already compiled.

The script `docker-php-ext-enable` allows to enabling of a pre-compiled and installed extension (bundled or PECL).

IMHO if PIE must know the bundled PHP extension is only to enable it if necessary for all SAPI installed.

Enable an extension is very different from various installations.

**Comment by faizanakram99** [comment]
fwiw, amazon linux often ships php without sodium initially for a few months and makes sodium available later. Same goes with ext-zip

**Comment by JoshuaBehrens** [comment]
*TL;DR* I would have a simpler time, when `pie` would be able handle all "well-known" extension as well, so `pie install` can be the equivalent to `composer install` / `nvm use` to solve all my dependency issues on any major OS where software nowadays is run.

<details>
<summary>
Story: I just want to tell a story about a task to add more examples, how pie can be a solution for all of us. You can ignore it for the discussion. Only helpful to collect problems of peoples everyday hussle
</summary>

 I stumbled upon this while moving from a docker based CI working with alpine Linux. We've reached certain limits on alpine Linux and want to move some parts to a debian based image. Changing all the distro specifics is not really easy. Moving from `php:8.3-cli-alpine` to `php:8.3-cli-bookworm` really grinded my gears. On alpine the aliases were fine, on debian not so much: `apt install -yy git curl bash make git composer patch php-bcmath php-dom php-simplexml php-sodium php-fileinfo php-ctype php-gd php-xml php-intl php-pdo php-pdo_mysql php-session` worked perfectly fine one alpine (obviously not using aptitude ...) but with a php8.3 tagged image I still get this output:

<details>
<summary>
aptitude output trying to resolve aliases to php8.2
</summary>

```
Package php-gd is a virtual package provided by:
  php8.2-gd 8.2.28-1~deb12u1 [Not candidate version]
Package php-xml is a virtual package provided by:
  php8.2-xml 8.2.28-1~deb12u1 [Not candidate version]
Package php-intl is a virtual package provided by:
  php8.2-intl 8.2.28-1~deb12u1 [Not candidate version]
Package php-pdo is a virtual package provided by:
  php8.2-common 8.2.28-1~deb12u1 [Not candidate version]
Package php-dom is a virtual package provided by:
  php8.2-xml 8.2.28-1~deb12u1 [Not candidate version]
Package php-bcmath is a virtual package provided by:
  php8.2-bcmath 8.2.28-1~deb12u1 [Not candidate version]
Package php-ctype is a virtual package provided by:
  php8.2-common 8.2.28-1~deb12u1 [Not candidate version]
Package php-fileinfo is a virtual package provided by:
  php8.2-common 8.2.28-1~deb12u1 [Not candidate version]
Package php-simplexml is a virtual package provided by:
  php8.2-xml 8.2.28-1~deb12u1 [Not candidate version]
E: Package 'php-bcmath' has no installation candidate
E: Package 'php-dom' has no installation candidate
E: Package 'php-simplexml' has no installation candidate
E: Unable to locate package php-sodium
E: Package 'php-fileinfo' has no installation candidate
E: Package 'php-ctype' has no installation candidate
E: Package 'php-gd' has no installation candidate
E: Package 'php-xml' has no installation candidate
E: Package 'php-intl' has no installation candidate
E: Package 'php-pdo' has no installation candidate
E: Unable to locate package php-pdo_mysql
E: Unable to locate package php-session
```

</details>

It is totally fine, that I have a hassle moving to a different distro. That specific task is not your issue. Without the core packages as part of the bundle I am not able to use `pie` as a solution to an OS agnostic approach. It is great, that composer is integrated in a way to automatically detect the dependencies. If it does not work for core ext though, why would I use `pie` for the others? I already have to run `brew`/`apk`/`yum`/`apt` for several extension for each environment next to `pecl` and `composer` to setup the community parts. Why would I want to throw in yet another tool, when it does not simplify things? People just throw in Laravel Herd because it brings it most of the well-known parts to not have issues setting the dev setup up. Something like this just for servers / CI and more part of the core-team and less at the Laravel-team would be so great.

If I just can pull out an alpine or debian image from a trusted and updated source I do not have to provide an updated docker image myself just to keep it updated as well all the time with the project specifica with a custom built php. I can do this to optimize processes but not for a quick entry.
</details>

**Comment by asgrim** [comment]
> *TL;DR* I would have a simpler time, when `pie` would be able handle all "well-known" extension as well, so `pie install` can be the equivalent to `composer install` / `nvm use` to solve all my dependency issues on any major OS where software nowadays is run.

Yes, that is my eventual goal 👍

**Comment by JoshuaBehrens** [comment]
Then I will be glad to challenge it in prod :D
